fsproxy/src/test/java/no/unit/identityservice/fsproxy/FsMock.java

```diff
@@ -0,0 +1,398 @@
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.time.Year;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import no.unit.identityservice.fsproxy.model.course.FsCourse;
+import no.unit.identityservice.fsproxy.model.course.FsCourseItemContainingCourseContainer;
+import no.unit.identityservice.fsproxy.model.course.FsCoursesSearchResult;
+import no.unit.identityservice.fsproxy.model.course.FsSemester;
+import no.unit.identityservice.fsproxy.model.course.FsSubject;
+import no.unit.identityservice.fsproxy.model.person.FsIdNumber;
+import no.unit.identityservice.fsproxy.model.person.FsIdSearchResult;
+import no.unit.identityservice.fsproxy.model.person.FsPerson;
+import no.unit.identityservice.fsproxy.model.person.FsPersonSearchResponse;
+import no.unit.identityservice.fsproxy.model.person.NationalIdentityNumber;
+import no.unit.identityservice.fsproxy.model.staffperson.FsCourseActivity;
+import no.unit.identityservice.fsproxy.model.staffperson.FsRoleToStaffPerson;
+import no.unit.identityservice.fsproxy.model.staffperson.FsRolesToPersonSearchResult;
+import no.unit.identityservice.fsproxy.model.staffperson.FsUriToCourseActivity;
+import no.unit.identityservice.fsproxy.model.staffperson.FsUriToCourseActivityContainer;
+import no.unit.nva.stubs.WiremockHttpClient;
+
+public class FsMock {
+
+    public static final String BIRTHDATE_IDENTIFIER = "fodselsdato0";
+    public static final String PERSONAL_NUMBER_IDENTIFIER = "personnummer0";
+    public static final String PERSON_ROLES = "/personroller";
+    public static final String SEMESTER_YEAR_PATH = "semester.ar";
+    public static final Integer CURRENT_YEAR = Year.now().getValue();
+    public static final Integer NEXT_YEAR = CURRENT_YEAR + 1;
+    private static final String PERSON_PATH = "/personer";
+    private static final String STUDENT_TEACHING_PATH = "/studentundervisning";
+    private static final String TEACHING_SEMESTER_AR_PATH = "undervisning.semester.ar";
+    private static final String FS_PERSON_ID_NUMBER_PATH = "person.personlopenummer";
+    private static final String DB_IDENTIFIER = "dbId";
+    private static final String LIMIT_IDENTIFIER = "limit";
+    private static final String LIMIT_VALUE = "0";
+    private final Map<NationalIdentityNumber, FsPerson> personEntries;
+    private final Map<FsIdNumber, List<FsRoleToStaffPerson>> personRoles;
+    private final Map<NationalIdentityNumber, List<FsCourse>> coursesForStudents;
+    private final Map<FsUriToCourseActivity, FsCourseActivity> courseActivities;
+    private final Map<FsRoleToStaffPerson, FsUriToCourseActivity> coursesToRoles;
+
+    private WireMockServer server;
+    private URI fsHostUri;
+    private HttpClient httpClient;
+
+    public FsMock() {
```

Comment:
This class is rather large and it does TWO things:
It generates / manages mock data AND it creates stub endpoints. You should consider splitting it into two classes.